test/common/mongo/proxy_test.cc

```diff
@@ -64,23 +67,98 @@ class MongoProxyFilterTest : public testing::Test {
 
     EXPECT_CALL(log_manager_, createAccessLog(_)).WillOnce(Return(file_));
     access_log_.reset(new AccessLog("test", log_manager_));
-    filter_.reset(new TestProxyFilter("test.", store_, runtime_, access_log_));
+  }
+
+  void initializeFilter() {
+    filter_.reset(
+        new TestProxyFilter("test.", store_, runtime_, access_log_, fault_config_, dispatcher_));
     filter_->initializeReadFilterCallbacks(read_filter_callbacks_);
     filter_->onNewConnection();
   }
 
+  void setupDelayFault(uint64_t duration_ms, uint32_t delay_percent, bool enable_fault) {
+    const std::string json_fault_config_template = R"EOF(
+    {{
```

Comment:
why the double { ?

Reply:
this is to make fmt::format happy :) (it's happy now)